cmd/gcp-controller-manager/csr_approver_test.go

```diff
@@ -432,6 +433,11 @@ func TestValidators(t *testing.T) {
 				goodCase(b, c)
 				b.ips = []net.IP{net.ParseIP("1.2.3.5")}
 			},
+			// Not matching DNS.
+			func(b *csrBuilder, c *controllerContext) {
+				goodCase(b, c)
+				b.dns = []string{"i1.z0.c.p1.internal"}
```

Comment:
Can you add a test case for DNS name without a zone? When does that happen?

Reply:
Added.